src/Workspaces/Core/Portable/Workspace/Solution/SolutionState.SymbolToProjectId.cs

```diff
@@ -14,6 +16,58 @@ internal partial class SolutionState
     {
         /// <inheritdoc cref="Solution.GetOriginatingProjectId"/>
         public ProjectId? GetOriginatingProjectId(ISymbol? symbol)
+        {
+            if (symbol == null)
+                return null;
+
+            var projectId = GetOriginatingProjectIdWorker(symbol);
+
+            // Validate some invariants we think should hold.  We want to know if this breaks, which indicates some part
+            // of our system not working as we might expect.  If they break, create NFWs so we can find out and
+            // investigate.
+
+            if (SymbolKey.IsBodyLevelSymbol(symbol))
```

Comment:
@jasonmalinowski i moved this to a non-impactful NFW throwing scenario instead.  Basically, ithink the *check* has high value, but i agree it is not necessary for correctness, and things are nicer/cleaner if we just go through teh single IAssemblySymbol path.  But thsi check will help us find out if there are parts of our system that are violating our presumed invariants.